### PR TITLE
fix: remove double quotes from env input variables

### PR DESCRIPTION
**Pull request by HimanshuJha2000**
# Description

Fixes https://github.com/devtron-labs/devtron/issues/1685
Fixes the double quotes issue occuring in input environment variables.
## How Has This Been Tested?
Tested manually for all possible scenarios

## Checklist:

* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [x] I have performed a self-review of my own code.
* [x] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: scriptExecutor.go [hunk 1]
```diff
@@ -196,7 +200,6 @@ func RunScriptsInDocker(executionConf *executionConf) (map[string]string, error)
 		log.Println(util.DEVTRON, err)
 		return nil, err
 	}
-	fmt.Println(dockerRunCommand)
```
Comment (subhashish-devtron) [inline scriptExecutor.go:199]: why is this removed?

File: scriptExecutor.go [hunk 2]
```diff
@@ -271,3 +274,36 @@ func buildDockerRunCommand(executionConf *executionConf) (string, error) {
 	return finalScript, nil
 
 }
+
+// Writes input vars to env file
+func writeToEnvFile(envMap map[string]string, filename string) error {
+	content, err := marshal(envMap)
+	if err != nil {
+		return err
```
Comment (subhashish-devtron) [inline scriptExecutor.go:282]: since we control this logic now, can we add proper error messages here? Applies at other places as well.

File: scriptExecutor.go [hunk 3]
```diff
@@ -271,3 +274,36 @@ func buildDockerRunCommand(executionConf *executionConf) (string, error) {
...
+	_, err = file.WriteString(content + "\n")
+	if err != nil {
+		return err
+	}
+	file.Sync()
+	return err
+}
+
+// Filters values of env variables on the basis of type and inserts to slice of strings
+func marshal(envMap map[string]string) (string, error) {
```
Comment (subhashish-devtron) [inline scriptExecutor.go:298]: when is this method going to return a non nil error?

File: scriptExecutor.go [hunk 4]
```diff
@@ -271,3 +274,36 @@ func buildDockerRunCommand(executionConf *executionConf) (string, error) {
...
+	lines := make([]string, 0, len(envMap))
+	for k, v := range envMap {
+		if d, err := strconv.Atoi(v); err == nil {
```
Comment (subhashish-devtron) [inline scriptExecutor.go:301]: can we make this more comprehensible by keeping a boolean flag `isValidInteger`

File: scriptExecutor.go [hunk 5]
```diff
@@ -271,3 +274,36 @@ func buildDockerRunCommand(executionConf *executionConf) (string, error) {
...
+			lines = append(lines, fmt.Sprintf(`%s=%d`, k, d))
+		} else {
+			lines = append(lines, fmt.Sprintf(`%s=%s`, k, v))
+		}
+	}
+	sort.Strings(lines)
```
Comment (subhashish-devtron) [inline scriptExecutor.go:307]: why is there a need to sort?

File: scriptExecutor.go [hunk 6]
```diff
@@ -271,3 +274,36 @@ func buildDockerRunCommand(executionConf *executionConf) (string, error) {
...
+	return strings.Join(lines, "\n"), nil
```
Comment (subhashish-devtron) [inline scriptExecutor.go:308]: minor: can have a const `newLineCharacter`

**Review by subhashish-devtron — COMMENTED** [review]
let's add unit tests for `writeToEnvFile()`

File: scriptExecutor.go (hunk 3 above)
Comment (HimanshuJha2000) [reply scriptExecutor.go:298]: There was only one case where it could have been a non-nil err which was via 
`if d, err := strconv.Atoi(v); err == nil `
But as we're moving this to `isValidInteger` we no longer need to return any error

File: scriptExecutor.go (hunk 1 above)
Comment (HimanshuJha2000) [reply scriptExecutor.go:199]: Added it back

File: scriptExecutor.go (hunk 2 above)
Comment (HimanshuJha2000) [reply scriptExecutor.go:284]: done

File: scriptExecutor.go (hunk 4 above)
Comment (HimanshuJha2000) [reply scriptExecutor.go]: done

File: scriptExecutor.go (hunk 6 above)
Comment (HimanshuJha2000) [reply scriptExecutor.go]: done

**Comment by HimanshuJha2000** [comment]
> let's add unit tests for `writeToEnvFile()`

added the UT

**Review by subhashish-devtron — COMMENTED** [review]
LGTM!

File: scriptExecutor.go [hunk 7]
```diff
@@ -271,3 +275,48 @@ func buildDockerRunCommand(executionConf *executionConf) (string, error) {
...
+	if err != nil {
+		log.Println(util.DEVTRON, "error while writing env values to file ", err)
+		return err
+	}
+	file.Sync()
+	return err
+}
+
+// Filters values of env variables on the basis of type and inserts to slice of strings
+func marshal(envMap map[string]string) (string, error) {
```
Comment (kartik-579) [inline scriptExecutor.go]: marshal does not seem suitable name for the work this method is doing, can we have name it something more aligned with the work it is doing?

File: scriptExecutor.go [hunk 8]
```diff
@@ -271,3 +275,48 @@ func buildDockerRunCommand(executionConf *executionConf) (string, error) {
...
+	lines := make([]string, 0, len(envMap))
+	for k, v := range envMap {
+		if d, ok := isValidInteger(v); ok {
+			lines = append(lines, fmt.Sprintf(`%s=%d`, k, d))
+		} else {
+			lines = append(lines, fmt.Sprintf(`%s=%s`, k, v))
+		}
+	}
+	//sort.Strings(lines)
```
Comment (kartik-579) [inline scriptExecutor.go]: can remove this

**Review by kartik-579 — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 9]
```diff
@@ -12,7 +12,7 @@ COPY . .
 RUN CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o /go/bin/cirunner
 
 
-FROM docker:20.10.17-dind
+FROM docker:20.10.24-dind
```
Comment (kartik-579) [inline Dockerfile:15]: remove this change please

**Review by kartik-579 — APPROVED** [review]
